### ISSUE TITLE
Internal error with sized types if the target type of a constructor is an alias

### ISSUE THREAD

**Issue by andy-morris** [agda/agda#4049]
With Agda 2.6.0.1, the following code gives an internal error pointing to `src/full/Agda/TypeChecking/Polarity.hs:307`.

```agda
open import Agda.Builtin.Size

data A : Size → Set
B = A

data A where a : ∀ i → B i
```

The code seems to assume that the target type of a constructor is a literal occurence of the parent datatype, which of course may not be true in all cases.

**Comment by gallais** [comment]
The error is raised in a function whose precondition is that the input type is
reduced. Evidently it is called somewhere with this condition violated.

Edit: cannot reproduce in 2.6.1 (I should have started with that!).

**Comment by gallais** [comment]
Most likely fixed by the fix for #3640
